### DISCUSSION TITLE
Initial permissions for administrators

### DISCUSSION THREAD

**Discussion by rtrvrtg** [MukurtuCMS/Mukurtu-CMS#78] [Alpha testing feedback]
Hi, I've got a couple of Mukurtu 4 environments up and running for development and testing. One thing I've noticed is that the administrator permissions don't let you do much on a fresh install. None of the Node permissions for the Mukurtu Administrator have been enabled, meaning that only UID 1 can create anything.

Here's an example screenshot my colleague took: 
![MicrosoftTeams-image (11)](https://github.com/MukurtuCMS/Mukurtu-CMS/assets/587242/6942bef7-1eb8-4a3d-bb10-145747e08b1a)

There's probably something that I'm missing here, whether it be a step during the setup process, another module that controls content permissions (Organic Groups maybe?), or that M4 is still under development and the admin permissions haven't been got around to yet. Nevertheless, we'd like to try and test the content creation/import process, so I'm wondering whether it's safe to just turn everything on and see how it goes. I'm also happy to have a go at trying to configure content permissions for the Mukurtu Administrator role if you like but would like to know what sort of access to core Drupal functionality that you have in mind for the role.

Please let me know if you need more info about our situation, or if we can be of any assistance. Thanks!

**Comment by taylor-steve** [comment]
Hi! 

Some of this is intentional, while other parts are likely waiting for another pass by us.

In Mukurtu CMS (and this isn't very clear yet in v4, all the documentation is on the v3 site or https://mukurtu.org/) administrator roles aren't intended to have access to content permissions by that role alone. They are meant to setup some site configuration, create communities, and grant the "community manager" role to users. From there, the community managers create protocols and hand out the relevant protocol roles (e.g., protocol steward). Those protocol roles are what grant content/media permissions to users. On smaller sites or test sites, admins might grant those community & protocol roles to themselves.

We try and respect the permissions, so you are free to create your own roles/modify permissions to match the needs for your site. At the stage we're at right now, the community and protocol level permissions (dashboard/group/permissions/community/community, dashboard/group/permissions/protocol/protocol) are likely to work as described. Some site permissions may not yet, at least in certain contexts.

As for safety, this Mukurtu CMS v4 is under active development and we're not at a point of stability. Things get rewritten or left in an incomplete state quite often. For local testing you're of course more than welcome to try anything, but the software isn't yet suitable for production use, and everything is subject to change at any time.

Happy to hear you're taking a look!

**Comment by michael-wynne-wsu** [comment]
One additional note - you will need to create a community and protocol (as shown in that screenshot) before you can create any content (digital heritage, collection, etc) since all content is protocol-aware/required.

I just checked on the latest gitpod build and a Mukurtu Administrator (not UID 1) can successfully go through those steps: create community > create protocol > create digital heritage.

If you did already create those groups, you will also need to assign other users permissions within them (in the same order, community > protocol) before those users can create content using those protocols.

**Comment by rtrvrtg** [comment]
@steve-taylor-wsu, @michael-wynne-wsu, thanks for your help! I've followed your instructions and confirmed that creating a community then protocols attached to it enables users within that community to start creating content. We're good to keep going with testing, so am happy to close this discussion off.

**Reply by michael-wynne-wsu** [reply]
We appreciate the testing and feedback! Do open up more tickets as you come across other items.
